activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -231,7 +233,7 @@ def size
       # loaded and you are going to fetch the records anyway it is better to
       # check <tt>collection.length.zero?</tt>.
       def empty?
-        if loaded?
+        if loaded? || @association_ids
```

Comment:
We talked about this briefly in the issue and think this should probably have the `!find_target?` check as well. Can you investigate why that may matter, and if it does, add a test for it?

Reply:
I think we don't need the `!find_target?` since that will be implicit if `@association_ids` is defined because it calls `size.zero?`.